### PR TITLE
#265 Searchbar reference select

### PR DESCRIPTION
**Pull request by qscgyjqscgyj**
Ref #265

### CODE REVIEW

File: src/components/SearchBar/SearchBarColumn/DateColumn/index.tsx [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import { Col } from 'antd';
 
 import { DatePicker } from 'src/components/DatePicker';
+import { SearchBarColumnDateTypeProps } from 'src/components/SearchBar/SearchBarColumn/types';
```
Comment (ir4y) [inline src/components/SearchBar/SearchBarColumn/DateColumn/index.tsx:4]: Why import is changed?

File: src/components/SearchBar/SearchBarColumn/index.tsx [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
     const { columnFilterValue } = props;
 
-    switch (columnFilterValue.column.type) {
-        case 'string':
-            return <StringColumn<T> {...(props as SearchBarColumnStringTypeProps)} />;
-        case 'date':
-            return <DateColumn<T> {...(props as SearchBarColumnDateTypeProps)} />;
-        default:
-            return null;
+    if (isStringColumnFilterValue(columnFilterValue)) {
```
Comment (ir4y) [inline src/components/SearchBar/SearchBarColumn/index.tsx:20]: Rid of type conversion and `as` operator

File: src/components/SearchBar/hooks.ts [hunk 3]
```diff
@@ -1,30 +1,69 @@
...
+    StringTypeColumnFilterValue,
+} from './types';
 
 export function useSearchBar(props: SearchBarProps): SearchBarData {
     const { columns } = props;
 
     const defaultFiltersValues = useMemo(() => {
-        return columns.map((column) => ({ column }));
-    }, [columns]);
+        return columns.map((column) => {
```
Comment (ir4y) [inline src/components/SearchBar/hooks.ts:27]: Try to read of `as` operator and type conversion

File: src/components/SearchBar/hooks.ts [hunk 4]
```diff
@@ -1,30 +1,69 @@
...
     const [columnsFilterValues, setColumnsFilterValues] = useState<ColumnFilterValue[]>(defaultFiltersValues);
 
     const onChangeColumnFilter = useCallback(
-        (value: DateColumnFilterValue | string, id: string) => {
-            setColumnsFilterValues((prevFilterValue) => {
-                const newFilterValue = [...prevFilterValue];
+        (value: ColumnFilterValue['value'], id: string) => {
+            setColumnsFilterValues((prevFilterValues) => {
+                const newFilterValues = [...prevFilterValues];
+                const newFilterValueIndex = newFilterValues.findIndex((v) => v.column.id === id);
```
Comment (ir4y) [inline src/components/SearchBar/hooks.ts:45]: Try to use
```
const newFilterValueIndex = newFilterValues.find((v) => v.column.id === id);
```

File: src/components/SearchBar/SearchBarColumn/StringColumn/hooks.ts [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import { useCallback } from 'react';
 
-import { SearchBarColumnStringTypeProps } from '../types';
```
[the remark this answers is not in the document]
Comment (qscgyjqscgyj) [reply src/components/SearchBar/SearchBarColumn/StringColumn/hooks.ts:3]: done

File: src/components/SearchBar/SearchBarColumn/index.tsx (hunk 2 above)
Comment (qscgyjqscgyj) [reply src/components/SearchBar/SearchBarColumn/index.tsx:15]: done

File: src/components/SearchBar/hooks.ts (hunk 3 above)
Comment (qscgyjqscgyj) [reply src/components/SearchBar/hooks.ts:25]: done

File: src/components/SearchBar/hooks.ts (hunk 4 above)
Comment (qscgyjqscgyj) [reply src/components/SearchBar/hooks.ts]: done

**Pull request merged**